test/jdk/com/sun/crypto/provider/DHKEM/Compliance.java

```diff
@@ -0,0 +1,317 @@
+        Asserts.assertEQ(enc.key().getAlgorithm(), "Generic");
+
+        kem.newDecapsulator(kpX.getPrivate(), null);
+        KEM.Decapsulator d = kem.newDecapsulator(kpX.getPrivate());
+        d.decapsulate(enc.encapsulation());
+        SecretKey dec = d.decapsulate(enc.encapsulation());
+        Asserts.assertTrue(Arrays.equals(enc.key().getEncoded(), dec.getEncoded()));
+
+        dec = d.decapsulate(enc.encapsulation(), 0, d.secretSize(), "AES");
+        Asserts.assertTrue(Arrays.equals(enc.key().getEncoded(), dec.getEncoded()));
+
+        Utils.runAndCheckException(
+                () -> d.decapsulate(null),
+                NullPointerException.class);
+        Utils.runAndCheckException(
+                () -> d.decapsulate(enc.encapsulation(), -1, 12, "AES"),
+                IndexOutOfBoundsException.class);
+        Utils.runAndCheckException(
+                () -> d.decapsulate(enc.encapsulation(), 0, d.secretSize() + 1, "AES"),
+                IndexOutOfBoundsException.class);
+        Utils.runAndCheckException(
+                () -> d.decapsulate(enc.encapsulation(), 0, d.secretSize(), null),
+                NullPointerException.class);
+
+        KEM.Encapsulator e3 = kem.newEncapsulator(kpEC.getPublic());
+        KEM.Encapsulated enc2 = e3.encapsulate();
+        KEM.Decapsulator d3 = kem.newDecapsulator(kpX.getPrivate());
+        Utils.runAndCheckException(
+                () -> d3.decapsulate(enc2.encapsulation()),
+                DecapsulateException.class);
+
+        Utils.runAndCheckException(
+                () -> d3.decapsulate(new byte[100]),
+                DecapsulateException.class);
+    }
+
+    static class MySecureRandom extends SecureRandom {
+        final Random ran;
+
+        MySecureRandom(long seed) {
+            ran = new Random(seed);
+        }
+
+        @Override
+        public void nextBytes(byte[] bytes) {
+            ran.nextBytes(bytes);
+        }
+    }
+
+    // Same random should generate same key encapsulation messages
+    static void determined() throws Exception {
+        long seed = new Random().nextLong();
+        byte[] enc1 = calcDetermined(seed);
+        byte[] enc2 = calcDetermined(seed);
+        Asserts.assertTrue(Arrays.equals(enc1, enc2),
+                "Undetermined for " + seed);
+    }
+
+    static byte[] calcDetermined(long seed) throws Exception {
+        SecureRandom random = new MySecureRandom(seed);
+        KeyPairGenerator g = KeyPairGenerator.getInstance("XDH");
+        g.initialize(NamedParameterSpec.X25519, random);
+        PublicKey pk = g.generateKeyPair().getPublic();
+        KEM kem = KEM.getInstance("DHKEM");
+        kem.newEncapsulator(pk, random); // skip one
+        KEM.Encapsulator e = kem.newEncapsulator(pk, random);
+        byte[] enc1 = e.encapsulate().encapsulation();
+        byte[] enc2 = e.encapsulate().encapsulation();
+        Asserts.assertFalse(Arrays.equals(enc1, enc2));
```

Comment:
Another case,
KEM kem = KEM.getInstance("DHKEM");
   KEM.Encapsulator e1 = kem.newEncapsulator(pk, random);
KEM kem1 = KEM.getInstance("DHKEM");
   KEM.Encapsulator e2 = kem1.newEncapsulator(pk, random);
byte[] enc1 = e1.encapsulate().encapsulation();
byte[] enc2 = e2.encapsulate().encapsulation();
Asserts.assertFalse(Arrays.equals(enc1, enc2));

Can this case be added too?